server/routes/maintainingReports/coordinator.ts

```diff
@@ -49,6 +62,167 @@ export default class CoordinatorRoutes {
     })
   }
 
+  viewEditIncidentDetails: RequestHandler = async (req, res) => {
+    req.flash('changes') // clear out any cross-contaminatng data
+    const { reportId } = req.params
+    const newPrison = req.query['new-prison']
+    const systemToken = await this.authService.getSystemClientToken(res.locals.user.username)
+    const report = await this.reviewService.getReport(parseInt(reportId, 10))
+    const incidentLocationsInPersistedPrison = await this.locationService.getIncidentLocations(
+      systemToken,
+      report.agencyId
+    )
+    const { incidentDetails } = await this.reportDetailBuilder.build(res.locals.user.username, report)
+    const userInput = req.flash('coordinatorInputForEditIncidentDetails')
+    const input = firstItem(userInput)
+    const incidentDate = getIncidentDate(incidentDetails.incidentDate, input?.incidentDate)
+    const pageData = input || incidentDetails
+
+    let newPrisonDetails = null // newPrison refers to the one selected via the /edit-prison page
+    let incidentLocationsInNewPrison = null
+
+    if (newPrison) {
+      try {
+        newPrisonDetails = await this.locationService.getPrisonById(systemToken, newPrison.toString())
+        incidentLocationsInNewPrison = await this.locationService.getIncidentLocations(
+          systemToken,
+          newPrison.toString()
+        )
+      } catch (error) {
+        log.error(`User attempted to obtain details for prison ${newPrison}`)
+      }
+    }
+
+    const data = {
+      ...pageData,
+      reportId,
+      witnesses: input?.witnesses || report.form.incidentDetails.witnesses,
+      displayName: incidentDetails.offenderName,
+      incidentDate,
+      locations: incidentLocationsInNewPrison || incidentLocationsInPersistedPrison,
+      prison: newPrisonDetails || incidentDetails.prison,
+      newAgencyId: newPrison,
+    }
+
+    const errors = req.flash('errors')
+
+    return res.render('pages/coordinator/incident-details.njk', {
+      data,
+      errors,
+      showSaveAndReturnButton: false,
+      coordinatorEditJourney: true,
+    })
+  }
+
```

Comment:
const inputData object holds the data currently saved in the report, the new user input and also a boolean that determines if the two are different. The boolean is later used to filter out only the values  that have changed and only these will be  rendered in the next page in the sequence